### PR TITLE
Add OSS variants of artifact rake tasks

### PR DESCRIPTION
**Pull request by andrewvc**
These will omit the x-pack folder once it is merged in

### CODE REVIEW

File: rakelib/artifacts.rake [hunk 1]
```diff
@@ -62,13 +63,26 @@ namespace "artifact" do
...
+  def path_matches_exclude?(path, ex)
+    path == ex || (File.directory?(ex) && path =~ /^#{ex}\//)
+  end
+
+  def files(excluder=nil)
+    excluder ||= self.method(:exclude?)
     return @files if @files
     @files = package_files.collect do |glob|
-      Rake::FileList[glob].reject { |path| exclude?(path) }
+      Rake::FileList[glob].reject { |path| excluder.call(path) }
```
Comment (yaauie) [inline rakelib/artifacts.rake]: We can avoid an extra scope by passing the unbound method as a block:

~~~ ruby
      Rake::FileList[glob].reject(&excluder)
~~~

File: rakelib/artifacts.rake [hunk 2]
```diff
@@ -62,13 +63,26 @@ namespace "artifact" do
...
+    excludes.any? { |ex| path_matches_exclude?(path, ex) }
   end
 
-  def files
+  def exclude_oss?(path)
+    path_matches_exclude?(path, "x-pack" ) || exclude?(path)
+  end
+
+  def oss_excluder
+    @oss_excluder ||= self.method(:exclude_oss?)
```
Comment (yaauie) [inline rakelib/artifacts.rake]: while it works, passing around memoised unbounded methods isn't the most straight-forward to understand; let's make a note to take some time on cleaning this up when we're not under pressure timing-wise.

Comment (andrewvc) [reply rakelib/artifacts.rake]: Maybe I've been doing FP for too long but it doesn't seem weird to me? What would the cleanup be?

Comment (andrewvc) [reply rakelib/artifacts.rake]: I mean, we could make it an option `:exclude_oss` that the various methods are provided with, but I think they read the same. Passing a predicate seems just as readable IMHO

**Pull request closed without merging**